Bug 1072350 - Removing CPOWs used by Find events.

krizsa opened the pull request.

krizsa committed:
Bug 1072350 - Removing CPOWs used by Find events.

Snuffleupagus commented:
/botio-linux preview

Snuffleupagus commented:
With this patch applied, searching seems to work exactly the same both with/without e10s, which is great!

However, I noticed that with e10s enabled the following message is printed in the browser console:
```
A request for a CPOW wrapped chrome window failed for unknown reasons. PdfjsContentUtils.jsm:175:0
```
I'm not sure if that is related to this patch, or if it existed previously and was simply "uncovered" by the patch, but I figured I would mention it just in case. (Direct link to the code: [PdfjsContentUtils.jsm:175:0](https://github.com/mozilla/pdf.js/blob/master/extensions/firefox/content/PdfjsContentUtils.jsm#L175-L176).)

Snuffleupagus commented on extensions/firefox/content/PdfStreamConverter.jsm:
```diff
@@ -795,41 +810,28 @@ FindEventManager.prototype.bind = function() {
-    var type = this.types[i];
-    this.eventElement.addEventListener(type, this, true);
-  }
+  // We cannot directly attach listeners to for the find events
```
Nit: "... to for ..." seems like a typo.

krizsa replied:
That is weird. getChromeWindow and getFindBar from PdfjsContentUtils.jsm should never be called, in fact it's a bug that I have not removed them like I did for their counterpart from PdfjsChromeUtils.jsm ... Do you by any chance know who is the caller when that happens?

timvandermeij commented:
The commit also needs to be rebased to the current master.

Snuffleupagus commented:
@krizsa The only caller of [`getChromeWindow`](https://github.com/mozilla/pdf.js/blob/master/extensions/firefox/content/PdfStreamConverter.jsm#L66-L71) that I can find in the code is [PdfStreamConverter.jsm#L997](https://github.com/mozilla/pdf.js/blob/master/extensions/firefox/content/PdfStreamConverter.jsm#L997), or in your patch [PdfStreamConverter.jsm#L999](https://github.com/mozilla/pdf.js/pull/5749/files#diff-985966130e4e9f07e74f562c737dee81R999).
Unfortunately the addon debugger ignored my attempts to set breakpoints with e10s active, so I cannot confirm it, but according to a search that should be the only place.

krizsa commented:
@Snuffleupagus Thanks, that call was totally unnecessary I thought I have removed that line. Maybe it was lost when I converted the patch from mercurial to git. Anyway, I fixed it, so the problem should be fixed now, and I've also rebased the patch.

Snuffleupagus commented:
/botio-linux preview

yurydelendik commented:
Really good. That you for the patch.
